src/char/mapif.c

```diff
@@ -1848,31 +1849,36 @@ static int mapif_account_storage_load(int fd, int account_id)
 
 /**
  * Parses account storage load request from map server.
- * @packet 0x3010 [in] <account_id>.L
+  * @packet 0x3010 [in] <account_id>.L <storage_id>.W <storage_size>.W
  * @param  fd     [in] file/socket descriptor
  * @return 1 on success, 0 on failure.
  */
 static int mapif_parse_AccountStorageLoad(int fd)
 {
-	int account_id = RFIFOL(fd, 2);
+	int account_id = RFIFOL(fd, 2), storage_id = RFIFOW(fd, 6);
+	int storage_size = RFIFOW(fd, 8);
 
 	Assert_ret(fd > 0);
 	Assert_ret(account_id > 0);
+	Assert_ret(storage_id >= 0);
+	Assert_ret(storage_size > 0);
 
-	mapif->account_storage_load(fd, account_id);
+	mapif->account_storage_load(fd, account_id, storage_id, storage_size);
 
 	return 1;
 }
 
 /**
  * Parses an account storage save request from the map server.
- * @packet 0x3011 [in] <packet_len>.W <account_id>.L <struct item[]>.P
+ * @packet 0x3011 [in] <packet_len>.W <account_id>.L <storage_id>.L <struct item[]>.P
  * @param  fd     [in] file/socket descriptor.
  * @return 1 on success, 0 on failure.
  */
 static int mapif_parse_AccountStorageSave(int fd)
 {
-	int payload_size = RFIFOW(fd, 2) - 8, account_id = RFIFOL(fd, 4);
+	int payload_size = RFIFOW(fd, 2) - 10, account_id = RFIFOL(fd, 4);
+	short storage_id = RFIFOW(fd, 8);
```

Comment:
```suggestion
	int storage_id = RFIFOW(fd, 8);
```

Reply:
It is of 2 bytes, only, but as I see other are using int, changed it. done